catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactoryTest.java

```diff
@@ -1150,16 +1156,17 @@ private Filter generateTemporalFilter(JAXBElement<BinaryComparisonOpType> tempor
         QueryImpl frameworkQuery = (QueryImpl) queryFactory.getQuery(grr)
                 .getQuery();
         N temporal = null;
+        Filter queryFilter = ((QueryImpl) frameworkQuery.getFilter()).getFilter();
         if (classes.length > 1) {
-            assertThat(frameworkQuery.getFilter(), instanceOf(Or.class));
+            assertThat(queryFilter, instanceOf(Or.class));
             int i = 0;
-            for (Filter filter : ((Or) frameworkQuery.getFilter()).getChildren()) {
+            for (Filter filter : ((Or) queryFilter).getChildren()) {
                 assertThat(filter, instanceOf(classes[i++]));
```

Comment:
There should be an assertion that the length of `classes` is equal to the number of `queryFilter`'s children.

Reply:
I see where you are coming from but I'd argue your suggestion isn't necessary because if the length of classes is not equal to the number of `queryFilter`s then the test will fail anyways with some index out of range error.

Reply:
There will be no `IndexOutOfBoundsException` if there are more classes than `queryFilter` children. Besides that, an assertion failing is more informative and compliant with testing infrastructure than an `Exception` or `Error` thrown during a test. I'll make it a :pencil2:, though.